enrollment-server/src/main/java/com/wultra/app/enrollmentserver/impl/service/PowerAuthActivationCodeHandler.java

```diff
@@ -0,0 +1,103 @@
+import com.wultra.security.powerauth.client.model.error.PowerAuthClientException;
+import com.wultra.security.powerauth.client.model.response.GetApplicationConfigResponse;
+import com.wultra.security.powerauth.client.v4.PowerAuthClient;
+import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * Specialization of {@link DelegatingActivationCodeHandler} using application configuration from PowerAuth Server.
+ *
+ * @author Lubos Racansky, lubos.racansky@wultra.com
+ */
+@AllArgsConstructor
+@Slf4j
+class PowerAuthActivationCodeHandler implements DelegatingActivationCodeHandler {
+
+    private static final String ACTIVATION_CODE = "activation_code";
+
+    private final PowerAuthClient powerAuthClient;
+
+    private final ObjectMapper objectMapper = new ObjectMapper().configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
+
+    @Override
+    public String fetchDestinationApplicationId(final String targetApplicationId, final String sourceApplicationId, final List<String> activationFlags, final List<String> applicationRoles) throws ActivationCodeException {
+        try {
+            final GetApplicationConfigResponse applicationConfig = powerAuthClient.getApplicationConfig(sourceApplicationId);
+            return applicationConfig.getApplicationConfigs().stream()
+                    .filter(it -> it.getKey().equals(ACTIVATION_CODE))
+                    .findFirst()
+                    .map(ApplicationConfigurationItem::getValues)
+                    .map(this::convert)
+                    .map(ActivationCodeConfiguration::allowedTargetApplicationIds)
+                    .filter(allowedIds -> allowedIds.contains(targetApplicationId))
+                    .map(allowedIds -> targetApplicationId)
+                    .orElse(null);
+        } catch (PowerAuthClientException e) {
+            throw new ActivationCodeException("Fetching application configuration for ID: %s failed.".formatted(sourceApplicationId), e);
+        }
+    }
+
+    private ActivationCodeConfiguration convert(final List<Object> values) {
+        return values.stream()
+                .map(this::convert)
+                .filter(Objects::nonNull)
+                .filter(ActivationCodeConfiguration::isTypeOfCopy)
+                .findFirst()
+                .orElse(null);
+    }
+
+    private ActivationCodeConfiguration convert(final Object value) {
+        try {
+            return objectMapper.convertValue(value, new TypeReference<>() {});
+        } catch (IllegalArgumentException e) {
+            logger.warn("Unable to convert {}", value, e);
+            return null;
+        }
+    }
+
+    private record ActivationCodeConfiguration(List<String> allowedTargetApplicationIds, ActivationType type) {
+        boolean isTypeOfCopy() {
+            return type == ActivationType.COPY;
+        }
+    }
+
+    private enum ActivationType {
```

Comment:
There is a clash with existing `ActivationType` in restful-integration.

I suggest `ActivationTransferType`.